unic/ucd/bidi/src/bidi_mirrored.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2015 The Servo Project Developers.
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+//! Unicode *Bidi_Mirrored* Character Property.
+
+
+use std::convert;
+use std::fmt;
+
+
+mod data {
+    use unic_utils::CharDataTable;
+    pub const BIDI_MIRRORED_TABLE: CharDataTable<()> = include!("tables/bidi_mirrored.rsv");
+}
+
+
+/// Return whether the given character gets mirrored in Right-to-Left text (`Bidi_Mirrored`).
+pub fn is_bidi_mirrored(ch: char) -> bool {
+    BidiMirrored::of(ch).into()
+}
+
+
+use unic_char_property::{BinaryCharProperty, CharProperty, TotalCharProperty};
+
+
+/// Represents values of the Unicode character property
+/// [*Bidi_Mirrored*](http://www.unicode.org/reports/tr44/#Bidi_Mirrored).
+///
+/// The value is `ture` if the character is a "mirrored" character in bidirectional text, `false`
+/// otherwise.
+#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
+pub struct BidiMirrored(bool);
+
+
+impl BidiMirrored {
+    /// Boolean value of this character property.
+    pub fn bool(&self) -> bool {
+        self.0
+    }
+}
+
+
+impl CharProperty for BidiMirrored {
+    fn prop_abbr_name() -> &'static str {
+        "Bidi_M"
+    }
+
+    fn prop_long_name() -> &'static str {
+        "Bidi_Mirrored"
+    }
+
+    fn prop_human_name() -> &'static str {
+        "Bidi Mirrored"
+    }
+}
+
+
+impl TotalCharProperty for BidiMirrored {
+    fn of(ch: char) -> Self {
+        Self::of(ch)
+    }
+}
+
+
+impl BinaryCharProperty for BidiMirrored {
+    #[inline]
+    fn bool(&self) -> bool {
+        self.bool()
+    }
+}
+
+
+impl convert::From<BidiMirrored> for bool {
+    fn from(bidi_m: BidiMirrored) -> bool {
+        bidi_m.bool()
+    }
+}
+
+impl Default for BidiMirrored {
+    #[inline]
+    fn default() -> Self {
+        BidiMirrored(false)
+    }
+}
+
+
+// NOTE: This cannot be generalized at the moment and needs to be implemented for concrete types
+// indivisually. See <https://users.rust-lang.org/t/12884>.
+impl fmt::Display for BidiMirrored {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.human_name())
+    }
+}
+
+
+impl BidiMirrored {
+    /// Find the character *BidiMirrored* property value.
+    pub fn of(ch: char) -> Self {
+        use unic_utils::CharDataTable;
+        BidiMirrored(
+            CharDataTable::<()>::find(&data::BIDI_MIRRORED_TABLE, ch).is_some(),
```

Comment:
This _shouldn't_ be required, you should be able to just write `data::BIDI_MIRRORED_TABLE.contains(ch)`. If you can't, I messed something up and need to fix it with the new data table structure.